src/app/routes/connector-ui/contract-agreement-page/contract-agreement-transfer-dialog/contract-agreement-transfer-dialog.component.ts

```diff
@@ -88,37 +94,47 @@ export class ContractAgreementTransferDialogComponent implements OnDestroy {
     if (this.loading && !this.form.all.valid) {
       return;
     }
-    this.loading = true;
-    this.form.all.disable();
-
-    const value = this.form.value;
-    let request$: Observable<IdResponseDto>;
-    if (value.dataAddressType === 'Custom-Transfer-Process-Request') {
-      const request = this.buildCustomTransferRequest(value);
-      request$ = this.edcApiService.initiateCustomTransfer(request);
-    } else {
-      const request = this.buildTransferRequest(value);
-      request$ = this.edcApiService.initiateTransfer(request);
-    }
 
-    request$
-      .pipe(
-        finalize(() => {
-          this.loading = false;
-          this.form.all.enable();
-        }),
-      )
-      .subscribe({
-        next: (response) =>
-          this.close({
-            transferProcessId: response.id!,
-            contractId: this.data.contractId,
-          }),
-        error: (err) => {
-          this.notificationService.showError('Failed initiating transfer!');
-          console.error('Failed initiating transfer', err);
-        },
-      });
+    const confirmationDialogRef = this.confirmationDialog.open(
+      InitiateTransferConfirmTosDialogComponent,
+      {maxWidth: '30%'},
+    );
+
+    confirmationDialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+        this.loading = true;
```

Comment:
move the following code to another function to save on unnecessary nesting. this function does too much